### PR TITLE
Activate `noImplicitAny` and annotate everything with `AnnotateBecauseTodo` or similar

### PR DESCRIPTION
**Pull request by jimrandomh**
This PR continues from https://github.com/ForumMagnum/ForumMagnum/pull/6878 which adds type-annotations to most of `vulcan-forms`. It adds a set of types which are aliases for `any`, in `packages/lesswrong/lib/types/typeUtil.ts`: `AnyBecauseTodo`, `AnyBecauseObsolete`, `AnyBecauseIsInput`, and `AnyBecauseHard`. I then turned on the  `noImplicitAny` option in `tsconfig.json` and addressed every error that produced, annotating things with and casting things to `AnyBecauseTodo` if annotating them was even a tiny bit nontrivial.

(Note that this PR will rapidly accumulate implicit-any type errors as other things are merged, but that these should be pretty easy to address since they too can just be annotated with `AnyBecauseTodo`.)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204377219438897) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by jpaddison3** [comment]
Can you make the base branch of this be `typeAnnotateVulcanForms`? It will auto-update to be master once the first PR gets merged.

**Comment by oetherington** [comment]
> It will auto-update to be master once the first PR gets merged.

(Only if you manually delete the old branch sadly)

File: packages/lesswrong/lib/types/libraryTypes.ts [hunk 1]
```diff
@@ -6,3 +6,78 @@ declare module "deepmerge" {
...
+
+declare module "html-lexer" {
+  const HtmlLexer: AnyBecauseTodo
+  export default HtmlLexer
+}
+
+declare module "passport-auth0/lib/Profile" {
+  const Profile: AnyBecauseTodo
+  export default Profile;
+}
```
Comment (jpaddison3) [inline packages/lesswrong/lib/types/libraryTypes.ts:83]: Is this one not a regression? I thought this was typed

File: packages/lesswrong/lib/types/typeUtil.ts [hunk 2]
```diff
@@ -44,4 +44,39 @@ type FromPartial<T> = T extends Partial<infer U> ? U : never;
...
+ * expected to actually be used again.
+ */
+export type AnyBecauseObsolete = any
+
+/**
+ * Has Typescript type "any" because the developer spent more than 5 minutes
+ * trying to type this and was very frustrated. The developer promises they
+ * were not just being lazy.
+ */
+export type AnyBecauseHard = any
```
Comment (jpaddison3) [inline packages/lesswrong/lib/types/typeUtil.ts:80]: I'm glad you realized I was being serious

File: tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "lib": ["es2019", "dom"],
-    "noImplicitAny": false,
+    "noImplicitAny": true,
```
Comment (jpaddison3) [inline tsconfig.json:4]: Fuck. yes.

**Review by jpaddison3 — APPROVED** [review]
This is an absurd amount of work Jim. I am so excited to develop with this setting on. 🚀

**Pull request merged**